odoo/fields.py

```diff
@@ -1811,8 +1812,11 @@ def convert_to_column(self, value, record, values=None, validate=True):
             return None
         # Detect if the binary content is an SVG for restricting its upload
         # only to system users.
-        if value[:1] == b'P':  # Fast detection of first 6 bits of '<' (0x3C)
-            decoded_value = base64.b64decode(value)
+        if value[:1] in (b'P', 'P', b'<', '<'):  # Fast detection of first 6 bits of '<' (0x3C)
+            try:
+                decoded_value = base64.b64decode(value.translate(None, delete=b'\r\n'), validate=True)
```

Comment:
> @xmo-odoo `translate(None, delete=..)` won't work for strings AFAICT.

[What I originally reviewed](https://github.com/odoo/odoo/commit/eef64d7227a30071d2c1ac7e7e366f8bba455e70) only did anything if `b'P'` matched so we necessarily had bytes as input.

> Also I think this line is worth splitting the line for clarity, and then you can explain in a comment why the translate() call is there (i.e. handle RFC2045-wrapped b64)

My thinking was that it made much more sense to expound on this in the commit message since an explanation is also necessary for the `validate`, and the code is not fundamentally *tricky* (e.g. misleadingly simple-looking or something).

> PS: translate() will create another in-memory copy of the data. Probably fine for now, it's the easiest option.

I don't think we can avoid that. Even if we use a `bytearray` to remove things in-place, it'll need to first copy the source into the array before modifying it. `translate` was selected as a better alternative to doing two `replace` in a row (and thus having even more copies lying around) in order to strip both cr and lf.